### PR TITLE
T21 Forms

### PR DESCRIPTION
**Pull request by jenovateurs**
Hello!

* Type: new feature
* Link to issue:#21

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/phalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [ ] I wrote some tests for this PR
- [ ] I have updated the relevant CHANGELOG
- [ ] I have created a PR for the [documentation](https://github.com/phalcon/docs) about this change

Small description of change:

Thanks

### CODE REVIEW

File: src/Forms/Element/AbstractElement.php [hunk 1]
```diff
@@ -0,0 +1,494 @@
+<?php
+/**
```
Comment (Jeckerson) [inline src/Forms/Element/AbstractElement.php:3]: Add empty line between 1st line and comment. Also add `declare(strict_types=1);` in all php files as default.

File: src/Forms/Element/AbstractElement.php [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+
+    /**
+     * Adds a group of validators
+     *
+     * @param \Phalcon\Validation\ValidatorInterface[] validators
+     * @param bool                                     merge
+     */
+    public function addValidators(?array $validators, bool $merge = true):ElementInterface
+    {
+
```
Comment (Jeckerson) [inline src/Forms/Element/AbstractElement.php:111]:
```suggestion
```

File: src/Forms/Element/AbstractElement.php [hunk 3]
```diff
@@ -0,0 +1,494 @@
...
+    {
+        $this->messages->appendMessage($message);
+
+        return $this;
+    }
+
+    /**
+     * Clears element to its default value
+     */
+    public function clear():ElementInterface
```
Comment (Jeckerson) [inline src/Forms/Element/AbstractElement.php]:
```suggestion
    public function clear(): ElementInterface
```

File: src/Forms/Element/AbstractElement.php [hunk 4]
```diff
@@ -0,0 +1,494 @@
...
+    {
+        $form = $this->form;
+        $name = $this->name;
+        $value = $this->value;
+
+        if (gettype($form) == "object") {
```
Comment (Jeckerson) [inline src/Forms/Element/AbstractElement.php]: Instanceof or null check would be better.

File: src/Forms/Element/AbstractElement.php [hunk 5]
```diff
@@ -0,0 +1,494 @@
...
+            return $defaultValue;
+        }
+
+        return $attributes[attribute];
+    }
+
+    /**
+     * Returns the default attributes for the element
+     */
+    public function getAttributes():array
```
Comment (Jeckerson) [inline src/Forms/Element/AbstractElement.php]:
```suggestion
    public function getAttributes(): array
```

**Comment by niden** [comment]
Closing in favor of https://github.com/phalcon/phalcon/pull/292

**Pull request closed without merging**